dali/kernels/common/scatter_gather.h

```diff
@@ -93,15 +94,19 @@ class DLL_PUBLIC ScatterGatherBase {
     ranges_.resize(n);
   }
 
+  std::pair<size_t, size_t> BlockCountAndSize(const std::vector<CopyRange> &ranges) const;
```

Comment:
how about using `span<const CopyRange>` here and in line 103?

Reply:
The usage is limited to vector, so why compilcate?